File: handlerclaw/lib/features/notifications/domain/entities/notification_entity.dart
class NotificationEntity {
  final String id;
  final String notificationWebhookId;
  final String userId;
  final String userDeviceId;
  final String title;
  final String body;
  final Map<String, dynamic> data;
  final DateTime? triggeredAt;
  final String status;
  final String? fcmMessageId;
  final DateTime? sentAt;
  final DateTime? failedAt;
  final int retryCount;
  final String? errorMessage;
  final DateTime createdAt;
  final DateTime updatedAt;

  NotificationEntity({
    required this.id,
    required this.notificationWebhookId,
    required this.userId,
    required this.userDeviceId,
    required this.title,
    required this.body,
    required this.data,
    this.triggeredAt,
    required this.status,
    this.fcmMessageId,
    this.sentAt,
    this.failedAt,
    required this.retryCount,
    this.errorMessage,
    required this.createdAt,
    required this.updatedAt,
  });
}

File: api/src/modules/notification/domain/entities/notification.entity.ts
import type { NotificationStatus } from "../../../../lib/generated/prisma/client.js";

export type INotification = {
  id: string;
  notification_webhook_id: string;
  user_id: string;
  user_device_id: string;
  title: string;
  body: string;
  data: unknown;
  triggered_at: Date | null;
  status: NotificationStatus;
  fcm_message_id: string | null;
  sent_at: Date | null;
  failed_at: Date | null;
  retry_count: number;
  error_message: string | null;
  created_at: Date;
  updated_at: Date;
};
